azure/src/main/java/org/apache/iceberg/azure/AzureProperties.java

```diff
@@ -67,6 +83,16 @@ public AzureProperties(Map<String, String> properties) {
     if (properties.containsKey(ADLS_WRITE_BLOCK_SIZE)) {
       this.adlsWriteBlockSize = Long.parseLong(properties.get(ADLS_WRITE_BLOCK_SIZE));
     }
+    this.refreshCredentialsEndpoint = properties.get(REFRESH_CREDENTIALS_ENDPOINT);
+    this.refreshCredentialsEnabled =
+        PropertyUtil.propertyAsBoolean(properties, REFRESH_CREDENTIALS_ENABLED, true);
+    Map<String, String> credentialProviderProperties = Maps.newHashMap(properties);
```

Comment:
this can be moved into the `if` statement

Reply:
Moved the initialisation to if block.